modules/account-lib/src/coin/xrp/keyPair.ts

```diff
@@ -9,5 +13,115 @@ export class KeyPair extends Secp256k1ExtendedKeyPair {
    */
   constructor(source?: KeyPairOptions) {
     super(source);
+    if (!source) {
+      this.hdNode = HDNode.fromSeedBuffer(randomBytes(512 / 8));
+    } else if (isSeed(source)) {
+      this.hdNode = HDNode.fromSeedBuffer(source.seed);
+    } else if (isPrivateKey(source)) {
+      this.recordKeysFromPrivateKey(source.prv);
+    } else if (isPublicKey(source)) {
+      this.recordKeysFromPublicKey(source.pub);
+    } else {
+      throw new Error('Invalid key pair options');
+    }
+
+    if (this.hdNode) {
+      this.keyPair = this.hdNode.keyPair;
+      this.keyPair.compressed = this.hdNode.compressed;
+    }
```

Comment:
Can this be moved to `Secp256k1ExtendedKeyPair`?

Reply:
In eth.keypair, the implementation is different than other coins. So i dont think we should move it to Secp256k1ExtendedKeyPair.

Reply:
I have changed this in a separate commit. kindly check